openwisp_monitoring/device/api/views.py

```diff
@@ -71,7 +83,23 @@ def get_charts_args_rewrite(view, request, pk):
     return (pk,)
 
 
-class DeviceMetricView(MonitoringApiViewMixin, GenericAPIView):
+class DeviceKeyAuthenticationMixin(object):
```

Comment:
I found [`DevicePermission` class in openwisp-controller](https://github.com/openwisp/openwisp-controller/blob/c7f1cbd7d4fcdb197859be2908102c901ecfc7e9/openwisp_controller/geo/api/views.py#L34-L40). But, it will always require device.key in the API request.